Engine/ResourceManagement/Importer/SceneManager.cpp

```diff
@@ -0,0 +1,264 @@
+		UUIDS_pairs[original] = UUID;
+	}
+	if (prefab == nullptr)
+	{
+		GameObject * missing_prefab = App->scene->CreateGameObject();
+		missing_prefab->name = "Missing prefab";
+		return missing_prefab;
+	}
+	GameObject * instance = prefab->Instantiate(App->scene->GetRoot(), &UUIDS_pairs);
+	Config transform_config;
+	config.GetChildConfig("Transform", transform_config);
+	instance->transform.Load(transform_config);
+return instance;
+}
+
+bool SceneManager::SaveModifiedPrefabComponents(Config & config, GameObject * gameobject_to_save) const
+{
+
+	bool modified = false;
+	if (gameobject_to_save->transform.modified_by_user)
+	{
+		Config transform_config;
+		gameobject_to_save->transform.Save(transform_config);
+		config.AddChildConfig(transform_config, "Transform");
+		modified = true;
+	}
+	std::vector<Config> gameobject_components_config;
+	for (auto & component : gameobject_to_save->components)
+	{
+		if (component->modified_by_user || component->added_by_user)
+		{
+			Config component_config;
+			component->Save(component_config);
+			gameobject_components_config.push_back(component_config);
+			modified = true;
+		}
+	}
+	if (modified)
+	{
+		config.AddChildrenConfig(gameobject_components_config, "Components");
+		config.AddUInt(gameobject_to_save->UUID, "UUID");
+	}
+	return modified;
+}
+
+void SceneManager::LoadPrefabModifiedComponents(const Config & config) const
+{
+
+	GameObject * prefab_child = App->scene->GetGameObject(config.GetUInt("UUID", 0));
+	if (config.config_document.HasMember("Transform"))
+	{
+		Config transform_config;
+		config.GetChildConfig("Transform", transform_config);
+
+		prefab_child->transform.Load(transform_config);
+	}
+
+	std::vector<Config> prefab_components_config;
+	config.GetChildrenConfig("Components", prefab_components_config);
+	for (auto & component_config : prefab_components_config)
+	{
+		uint64_t component_type_uint = component_config.GetUInt("ComponentType", 0);
+		assert(component_type_uint != 0);
+
+		uint64_t UUID = component_config.GetUInt("UUID", 0);
+
+		Component * component = prefab_child->GetComponent(static_cast<Component::ComponentType>(component_type_uint));
+		if (component != nullptr && component->UUID == UUID)
+		{
+			component->Load(component_config);
+			component->modified_by_user = true;
+		}
+		else 
+		{
+			Component::ComponentType component_type = static_cast<Component::ComponentType>(component_type_uint);
+			Component* created_component = nullptr;
+			if (component_type == Component::ComponentType::COLLIDER) 
+			{
+				ComponentCollider::ColliderType collider_type = static_cast<ComponentCollider::ColliderType>(component_config.GetUInt("ColliderType", 0));
+				created_component = prefab_child->CreateComponent(collider_type);
+			}else {
+				Component* created_component = prefab_child->CreateComponent(static_cast<Component::ComponentType>(component_type_uint));
+			}
+			created_component->Load(component_config);
+			created_component->added_by_user = true;
+		}
+	}
+}
```

Comment:
This class is wrong. You messed up with the merge because on develop is completely different now.

Reply:
This was a ghost file it wasn't imported in the project but was among the files